src/System.Private.ServiceModel/tools/test/SelfHostWcfService/MyX509CertificateValidator.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IdentityModel.Selectors;
+using System.IdentityModel.Tokens;
+using System.Security.Cryptography.X509Certificates;
+
+namespace WcfService
+{
+    public class MyX509CertificateValidator : X509CertificateValidator
+    {
+        string allowedIssuerName;
+
+        public MyX509CertificateValidator(string allowedIssuerName)
+        {
+            if (string.IsNullOrEmpty(allowedIssuerName))
+            {
+                throw new ArgumentNullException("[MyX509CertificateValidator] The string allowedIssuerName was null or empty.");
```

Comment:
NIT: Usually customary to add the parameter name to the exception when there's an ArgumentNullException e.g., 

```csharp
    throw new ArgumentException(message, "parameterName"); 
```